pkg/logql/evaluator.go

```diff
@@ -626,26 +635,29 @@ func binOpStepEvaluator(
 	// implementation of this StepEvaluator
 	var scopedErr error
 
-	return newStepEvaluator(func() (bool, int64, promql.Vector) {
+	return NewStepEvaluator(func() (bool, int64, StepResult) {
 		var (
 			ts       int64
 			next     bool
 			lhs, rhs promql.Vector
+			r        StepResult
 		)
-		next, ts, rhs = rse.Next()
+		next, ts, r = rse.Next()
```

Comment:
everywhere we call `Next()` can we give the result a better name than `r`, took me a minute to track down what `r` actually was